### PR TITLE
Set config for new bafu layer

### PR DESCRIPTION
**Pull request by alcapat**
Link with chsdi https://github.com/geoadmin/mf-chsdi3/pull/1501

### CODE REVIEW

**Comment by cedricmoullet** [comment]
@ltclm Could you please review and merge if ok ?

**Comment by ltclm** [comment]
cannot be merged, please resolve the conflicts @alcapat

File: conf/bafu.conf.part [hunk 1]
```diff
@@ -17,6 +17,91 @@ source src_ch_bafu_hydrologie-untersuchungsgebiete_stationen : def_searchable_fe
...
+}
+
+source src_ch_bafu_vec25-seen : def_searchable_features
+{
+    sql_db = bafu_dev
+    sql_query = \
+        SELECT bgdi_id::bigint as id \
+        , name as label \
+        , 'feature' as origin \
+        , remove_accents(bgdi_id::text||' '||gewaesserkennzahl::text||' '||seetyp::text||' '||name||' '||ausgleichsbecken::text||' '||reguliert::text||' '||seeflaeche_km2::text||' '||inhalt_see_mio_m3::text||' '||nutzinhalt_mio_m3::text||' '||tiefe_see_m::text||' '||hoehenlage_muem::text||' ' ||uferlaenge_m::text||' '||gwlnr) as detail \
```
Comment (ltclm) [inline conf/bafu.conf.part]: why do you add the bgdi_id to the searchable attributes?

File: conf/bafu.conf.part [hunk 2]
```diff
@@ -17,6 +17,91 @@ source src_ch_bafu_hydrologie-untersuchungsgebiete_stationen : def_searchable_fe
...
 }
 
+source src_ch_bafu_vec25-gewaessernetz_2000 : def_searchable_features
+{
+    sql_db = bafu_dev
+    sql_query = \
+        SELECT bgdi_id::bigint as id \
+        , name as label \
+        , 'feature' as origin \
+        , remove_accents(bgdi_id::text||' '||gwlnr||' '||gewissnr::text||' '||name||' '||objectval) as detail \
```
Comment (ltclm) [inline conf/bafu.conf.part]: bgdi_id should not be used as feature id (linkt to object?)  and should not be searchable, it is not stable.

File: conf/bafu.conf.part [hunk 3]
```diff
@@ -17,6 +17,91 @@ source src_ch_bafu_hydrologie-untersuchungsgebiete_stationen : def_searchable_fe
...
+}
+
+source src_ch_bafu_hydrologischer-atlas_flussgebiete : def_searchable_features
+{
+    sql_db = bafu_dev
+    sql_query = \
+        SELECT bgdi_id::bigint as id \
+        , name as label \
+        , 'feature' as origin \
+        , remove_accents(bgdi_id::text||' '||nummer::text||' '||name||' '||shape_area::text||' '||umfang::text) as detail \
```
Comment (ltclm) [inline conf/bafu.conf.part]: the same here, do we really want bgdi_id and shape_area and umfang as searchable attributes?
The attributes which are concatenated should be wrapped inside coalesce function:
```sql
remove_accents(bgdi_id::text||' '||nummer::text||' '||name||' '||shape_area::text||' '||umfang::text)
```
should be
```sql
remove_accents(coalesce(bgdi_id::text,' ')||' '||coalesce(nummer::text,' ')||' '||coalesce(name,' ')||' '||coalesce(shape_area::text,' ')||' '||coalesce(umfang::text,' ')) as detail
```

**Comment by alcapat** [comment]
@ltcm confilt resolved

**Comment by ltclm** [comment]
please remove the bgdi_id from detail fieds in:
https://github.com/geoadmin/service-sphinxsearch/pull/217/files#diff-f766c2af301c761519b9a6cfebc31e2cR129
https://github.com/geoadmin/service-sphinxsearch/pull/217/files#diff-f766c2af301c761519b9a6cfebc31e2cR146

apart from that looks good to me, thanks!

**Comment by alcapat** [comment]
bgdi_id remove and coalesce function added

**Comment by ltclm** [comment]
thanks

**Pull request merged**